src/multiple-choice/components/app.tsx

```diff
@@ -4,24 +4,31 @@ import { Authoring } from "./authoring";
 import { Runtime } from "./runtime";
 
 export const App = () => {
-  const [ aspectRatio, setAspectRatio ] = useState(1);
   const interactiveAPIConfig = {
     interactiveState: true,
     authoredState: true,
-    aspectRatio
   };
-  const { mode, authoredState, interactiveState, setInteractiveState, setAuthoredState } = useLARAInteractiveAPI(interactiveAPIConfig);
+  const { mode, authoredState, interactiveState, setInteractiveState, setAuthoredState, setHeight } = useLARAInteractiveAPI(interactiveAPIConfig);
   const container = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
-    // Measure the main app container on every re-render and update aspect ratio. Note that useLARAInteractiveAPI
-    // takes care of its updates and will notify LARA about it using iframe phone.
-    const width = container.current?.offsetWidth;
+  const calcHeight = () => {
     const height = container.current?.offsetHeight
-    if (width && width > 0 && height && height > 0) {
-      setAspectRatio(width / height);
+    if (height && height > 0) {
+      setHeight(height);
     }
-  });
+  };
+
+  // Update height on every re-render.
+  useEffect(calcHeight);
+
+  useEffect(() => {
+    // It's necessary to wrap calcHeight in a closure function. Note that calcHeight will be recreated on every render.
+    // This will break the cleanup function which would try to remove wrong reference. resizeHandler is created
+    // only once, as useEffect has empty dependencies list.
+    const resizeHandler = () => calcHeight();
+    window.addEventListener("resize", resizeHandler);
+    return () => window.removeEventListener("resize", resizeHandler);
+  }, []);
```

Comment:
I'd guess this will be a shared pattern between most of these components. So eventually this should be moved into the shared folder somehow.

Reply:
Right, most of the App component will be reused. I'll think how to do it. I guess it can be a hook actually.